### ISSUE TITLE
prohibitedKeys is sometimes an object instead of an array

### ISSUE THREAD

**Issue by joecarney** [strongloop/loopback-datasource-juggler#1646]
# Description/Steps to reproduce

# Link to reproduction sandbox

# Expected result

# Additional information

Opening /explorer and running a query works first time but subsequent times it crashes with the following error:

```
Error-logger request  : GET /email-campaigns: TypeError: prohibitedKeys.indexOf is not a function
    at Object.<anonymous> (/node_modules/loopback-datasource-juggler/lib/utils.js:356:42)
```

I added the following line of debug to /lib/utls:

```
    /**
     * Make sure prohibited keys are removed from the query to prevent
     * sensitive values from being guessed
     */
    console.log('prohibitedKeys', prohibitedKeys);
    if (
      prohibitedKeys &&
      prohibitedKeys.indexOf(this.key) !== -1
    ) {
      offendingKeys.push(this.key);
      this.remove();
      return;
    }
```

First time I run the query...

```
2018-10-27T09:12:39.322Z prohibitedKeys []
2018-10-27T09:12:39.322Z prohibitedKeys []
2018-10-27T09:12:39.324Z prohibitedKeys []
```

Second time I run the query...

```
2018-10-27T09:12:40.312Z prohibitedKeys []
2018-10-27T09:12:40.312Z prohibitedKeys []
2018-10-27T09:12:40.313Z prohibitedKeys {}
```

As you can see, prohibitedKeys is changing from an array to object.

I've added a check for an Array just so I can continue working on my project...

```
    /**
     * Make sure prohibited keys are removed from the query to prevent
     * sensitive values from being guessed
     */
    console.log('prohibitedKeys', prohibitedKeys);
    if (
      prohibitedKeys &&
      Array.isArray(prohibitedKeys) &&
      prohibitedKeys.indexOf(this.key) !== -1
    ) {
      offendingKeys.push(this.key);
      this.remove();
      return;
    }
```

However, I think this problem is deeper inside loopback.  

My current config...

```
node -v
v11.0.0
```

```
mysql -V
mysql  Ver 8.0.12 for osx10.14 on x86_64 (Homebrew)
```

```
node -e 'console.log(process.platform, process.arch, process.versions.node)'
npm ls --prod --depth 0 | grep loopback

darwin x64 11.0.0
├── loopback@3.23.2
├── loopback-boot@2.27.1
├── loopback-component-explorer@6.3.1
├── loopback-connector-mysql@5.3.1
├── loopback-graphql-server@1.1.8
```

**Comment by xuanyue202** [comment]
same problem after upgrading to 3.25.0

**Comment by raymondfeng** [comment]
Do you have any model settings with `hidden: {}` or `hiddenProperties: {}`?

**Comment by raymondfeng** [comment]
No worries, I found the issue. Will fix it now.

**Comment by raymondfeng** [comment]
Fix released in `loopback-datasource-juggler@4.1.1` and `loopback-datasource-juggler@3.25.1`
